packages/sdk-react/src/core/queries/useReceivables.ts

```diff
@@ -590,3 +590,31 @@ export const useReceivableContacts = (receivable_id: string) => {
 
   return useCounterpartContactList(receivable?.counterpart_id);
 };
+
+export const useMarkInvoiceAsPaid = (receivable_id: string) => {
+  const { i18n } = useLingui();
+  const { api, queryClient } = useMoniteContext();
+
+  return api.receivables.postReceivablesIdMarkAsPaid.useMutation(
+    {
+      path: {
+        receivable_id,
+      },
+    },
+    {
+      onSuccess: async () => {
+        await Promise.all([
+          api.receivables.getReceivables.invalidateQueries(queryClient),
+          api.receivables.getReceivablesId.invalidateQueries(
```

Comment:
I believe we don’t need double invalidation here in the onSuccess handler. Currently, both api.receivables.getReceivables and api.receivables.getReceivablesId are being invalidated. Since marking a specific invoice as paid (by receivable_id) updates the status of that single invoice, invalidating api.receivables.getReceivablesId should be sufficient

Reply:
I agree, good catch.